source/features/restore-file.tsx

```diff
@@ -53,27 +62,51 @@ async function restoreFile(progress: (message: string) => void, menuItem: Elemen
 		throw new Error('Restore failed: File too big');
 	}
 
-	let {pathname} = menuItem.previousElementSibling as HTMLAnchorElement;
-	// Check if file was deleted by PR
+	// Attempt to get details from the `Edit` button in the menu
+	const {pathname: blobOrEditPath} = menuItem.previousElementSibling as HTMLAnchorElement;
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars -- Error comes up even though route and pathToFile are prefixed with a _
+	let [owner, repo, _route, branch, ..._pathToFile] = blobOrEditPath.startsWith('/')
+		? blobOrEditPath.slice(1).split('/') : blobOrEditPath.split('/');
+
 	if (menuItem.closest('[data-file-deleted="true"]')) {
+		// The file was deleted by the PR, restore it from the base commit
 		progress('Undeleting…');
-		const [nameWithOwner, headBranch] = select('.head-ref')!.title.split(':');
-		pathname = `/${nameWithOwner}/new/${headBranch}?filename=${filePath}`;
+		// The `Edit` button doesn't exist if the file is deleted
+		const [nameWithOwner, prBranch] = select('.head-ref')!.title.split(':');
+		owner = nameWithOwner.split('/')[0];
+		repo = nameWithOwner.split('/')[1];
+		branch = prBranch;
 	} else {
+		// The file has been modified, restore it to its original state
 		progress('Committing…');
+		// In this case, the edit button exists, so we already have the required info
 	}
 
 	const content = file.text;
-	// This is either an `edit` or `create` form
-	const form = (await fetchDom(pathname, 'form.js-blob-form'))!;
-	form.elements.value.value = content; // Restore content (`value` is the name of the file content field)
-	form.elements.message.value = (form.elements.message as HTMLInputElement).placeholder
-		.replace(/^Create|^Update/, 'Restore');
-
-	const response = await pushForm(form);
-	if (!response.ok) {
-		throw new Error(response.statusText);
-	}
+	await api.v4mutation(`mutation {
+		createCommitOnBranch(input: {
+			branch: {
+				repositoryNameWithOwner: "${owner}/${repo}",
+				branchName: "${branch}"
+			},
+			expectedHeadOid: "${await getHeadReference()}",
+			fileChanges: {
+				additions: [
+					{
+						path: "${filePath}",
+						contents: "${btoa(content)}"
+					}
+				]
+			},
+			message: {
+				headline: "Restore ${filePath}"
+			}
+		}) {
+			commit {
+				oid
+			}
```

Comment:
Maybe I can remove this, we don't need anything from the server in response

Reply:
We just need to confirm whether it was successful. How are errors reported back? Does `v4mutation` throw in that case?

Reply:
`v4mutation` is a copy of `v4`, so it does throw an error. I'll try merging the two functions...

Reply:
Seems like [GraphQL does not allow returning 'nothing at all'](https://github.com/apollographql/graphql-tools/issues/277), but you can return a value that might be null. So I'm keeping this `commit { oid }` part.